api/envoy/extensions/geoip_providers/maxmind/v3/maxmind.proto

```diff
@@ -0,0 +1,32 @@
+syntax = "proto3";
+
+package envoy.extensions.geoip_providers.maxmind.v3;
+
+import "envoy/extensions/geoip_providers/common/v3/common.proto";
+
+import "xds/annotations/v3/status.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.geoip_providers.maxmind.v3";
+option java_outer_classname = "MaxmindProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/geoip_providers/maxmind/v3;maxmindv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+option (xds.annotations.v3.file_status).work_in_progress = true;
+
+// [#protodoc-title: MaxMind geolocation provider]
+
+// Configuration for the MaxMind geolocation provider.
+// [#extension: envoy.geoip_providers.maxmind]
+message MaxMindConfig {
+  string city_db_path = 1;
+
+  string isp_db_path = 2;
+
+  string anon_db_path = 3;
```

Comment:
I see these are validated with code for "at least one must be set" - is there other validation that could be performed here for *if* they are set? (e.g. is there a specific string format they should match?)
Also, field comments would be good, in particular highlighting that at least one of them must be set (and that more than one may be?)

Reply:
yes good point, i missed adding api level comments.

Reply:
Given that individual db_path fields are not required to be set, but at least one is required to be set with file path format, it may be sophisticated to write validate rule in protoc-gen-validate format, e.g. smth like [(validate.rules).string.max_len = 0 or ((validate.rules).string.min_len = 1  and (validate.rules).string.pattern=[$FILE_PATH_PATTERN])]. I moved the validation to the code config level and if file path value is invalid maxmind provider will crash not being able to open the file. @ravenblackx do you think we should move validation into proto level instead?

Reply:
I think whatever validation *can* be done at the proto level should be, because doing that doubles as additional documentation for the user, and most likely gives better error messages (certainly better than a crashing service).

Doing the "at least one must be set" part in the code makes sense because there's no proto annotation mechanism for that.

Reply:
added a regex for db path "either empty value or ends with .mmdb" and when proto validation fails, it gives this error message: 
```
Proto constraint validation failed (MaxMindConfigValidationError.IspDbPath: value does not match regex pattern \"^$|^.*\\\\.mmdb$\"): isp_db_path: \"/geoip2/Isp.exe
```
May be advanced for users to read such regexes, going to add a comment as well about expected db file format